contributors/design-proposals/secure_watch.md

```diff
@@ -0,0 +1,288 @@
+to somehow ask for all Secrets attached to all pods bound to a given node in a
+single LIST or WATCH request). We obviously want to be able to authorize all
+other types of requests (in particular GETs), so that the design has to be
+consistent in all those cases.
+
+ To solve this problem, we propose to introduce the idea of ```bulk watch ``` and
+```bulk get ``` (and in the future other bulk operations). This idea was already
+appearing in the community and now it seems to be a good time to proceed with it.
+
+ We would like the API to be generic enough so that it will be useful for many
+different usecases, not just for this particular one. As a result, below we are
+describing requirements that the solution has to meet to satisfy our needs:
+- a single bulk request has to support multiple resources (e.g. get a node and
+all pods associated with it)
+- if the bulk request filters asks only for object of a single resource type,
+the returned result should ideally be comparible with what is returned from
+the regular API
+
+TODO: What else do we require?
+
+
+# Detailed design
+
+ To make the API generic enough, one of our requirements is to make those bulk
+operations work across different resource types (e.g. watch pod A and secret S
+within a single watch call). [Though, this doesn't need to be supported by the
+initial implementation - this just have to be possible at the API level.]
+
+ In the current API, it is not possible to express a request that is spanning
+multiple different resource types. So as a result, we need to introduce completely
+new API to support it.
+
+ We will solve it by introducing a new dedicated API group:
+```
+/apis/bulk.k8s.io/
+```
+ that underneath will have a separate implementation. Initially we are going
+to support the following two operations:
+```
+POST /apis/bulk.k8s.io/get   <body defines filtering>
+POST /apis/bulk.k8s.io/watch   <body defines filtering>
+```
+ Later, we would like to add ability for bulk create/update/delete with a
+similar pattern, though it's not part of this design.
+
+TODO: We can't define filtering as part of path (as we do in all our APIs)
+because the filtering definition can potentially be long and exceed the allowed
+length for the request. Though, since we will define the filtering in the body
+of the request anyway, maybe those can be GET requests?
+
+
+### Filtering definition
+
+ We need to make our filtering mechanism to support different resource types at
+the same time. On the other hand, we would like it to be as consistent with all
+other Kubernetes APIs as possible. So we define the selector for bulk operations
+as following:
+
+```
+// TODO: Find a better name for it?
+type BulkSelector struct {
+	SingleTypeBulkSelector []SingleTypeBulkSelector
+}
+
+// TODO: Find a better name for it?
+type SingleTypeBulkSelector struct {
+	// FIXME: This should be meta.k8s.io/GroupVersionKind instead (or something
```

Comment:
We can just close the watch and recreate with new set of secrets. That's still much better that what we are doing now.
That said, I agree it would be useful, but I don't think this should be "no-starter" for us.